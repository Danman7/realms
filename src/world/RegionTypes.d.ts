export interface ScenarioRegion {
  id: string
  name: string
  border: string
  namePath: string
  connectedRegionIds: string[]
}

export interface GameRegion extends ScenarioRegion {
  population: number
}

export interface Scenario {
  name: string
  regions: ScenarioRegion[]
  width: number
  height: number
}

export interface TestScenario extends Scenario {
  regions: GameRegion[]
}
