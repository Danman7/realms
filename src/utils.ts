export const getRandomItemFromArray = <T>(arr: T[]): T | undefined => {
  if (arr.length === 0) {
    return undefined // Return undefined if array is empty
  }
  const randomIndex = Math.floor(Math.random() * arr.length) // Generate random index
  return arr[randomIndex] // Return item at random index
}
