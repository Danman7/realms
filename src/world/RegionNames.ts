export const RegionNames = [
  'Eldermont',
  'Ravenshire',
  'Silverwood',
  'Crimsonburg',
  'Ironcrest',
  'Shadowhaven',
  'Moonstone Duchy',
  'Emberfall',
  'Thornwood',
  'Starhaven',
  'Ravenshadow',
  'Goldenhelm',
  'Stormwind',
  'Verdant Vale',
  'Wyvernwood',
  'Frostford',
  'Moonshadow Duchy',
  'Dragonspire',
  'Ravenshadow',
  'Celestial Domain'
]
