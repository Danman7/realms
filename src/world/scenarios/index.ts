import { Mythosia } from 'src/world/scenarios/Mythosia'
import { TestScenario } from 'src/world/scenarios/TestScenario'

export { Mythosia, TestScenario }
