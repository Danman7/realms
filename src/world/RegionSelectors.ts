import { RootState } from 'src/redux'

export const getRegions = (state: RootState) => state.regions
