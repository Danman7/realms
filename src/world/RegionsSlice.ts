import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { RegionTypes } from 'src/world'

export type RegionsState = RegionTypes.GameRegion[]

export const initialState: RegionsState = []

export const regionsSlice = createSlice({
  name: 'regions',
  initialState,
  reducers: {
    populateRegions: (
      _state,
      action: PayloadAction<RegionTypes.ScenarioRegion[]>
    ) =>
      action.payload.map(scenarioRegion => ({
        ...scenarioRegion,
        population: 0
      }))
  }
})

export const RegionActions = regionsSlice.actions

export const RegionsRducer = regionsSlice.reducer
