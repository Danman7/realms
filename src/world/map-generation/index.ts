import { RegionTypes } from 'src/world'
import { createRandomBaseRegions } from 'src/world/map-generation/RegionGeneration'

export const generateMap = (amount: number): RegionTypes.Region[] => {
  const baseRegions = createRandomBaseRegions(amount)

  return baseRegions
}
