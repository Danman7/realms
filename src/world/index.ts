import * as RegionTypes from 'src/world/RegionTypes.d'
import * as RegionSelectors from 'src/world/RegionSelectors'
import * as Scenarios from 'src/world/scenarios'
import { Region } from 'src/world/components/Region'
import { Map } from 'src/world/components/Map'

import { RegionActions, RegionsRducer } from 'src/world/RegionsSlice'

export {
  Scenarios,
  RegionTypes,
  RegionSelectors,
  RegionActions,
  Region,
  Map,
  RegionsRducer
}
