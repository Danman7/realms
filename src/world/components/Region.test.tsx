import { render } from 'src/utils/testUtils'
import { Region, Scenarios } from 'src/world'

describe('Region component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Region key="1" {...Scenarios.TestScenario.regions[0]} />
    )

    expect(container).toMatchSnapshot()
  })
})
