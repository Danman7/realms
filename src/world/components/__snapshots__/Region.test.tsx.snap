// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Region component should render correctly 1`] = `
<div>
  <g>
    <path
      class="sc-bdfCDU jctcNe"
      d=""
    />
    <path
      d=""
      fill="none"
      id="Region 1-name-path"
    />
    <text
      class="sc-gsTDqH ffSeKL"
    >
      <textpath
        font-size="24"
        href="#Region 1-name-path"
        startoffset="50%"
        text-anchor="middle"
      >
        Region 1
      </textpath>
    </text>
  </g>
</div>
`;
